### PR TITLE
fix: free msg_env unconditionally

### PR DESCRIPTION
**Pull request by savonarola**

### CODE REVIEW

File: c_src/main.c [hunk 1]
```diff
@@ -86,23 +86,16 @@ erlfdb_future_cb(FDBFuture* fdb_future, void* data)
     cancelled = future->cancelled;
     enif_mutex_unlock(future->cancel_lock);
 
-    enif_mutex_lock(future->msg_lock);
```
Comment (savonarola) [inline c_src/main.c:89]: The lock seems useless—it saves only from concurrent `erlfdb_future_cb` invocations, which the fdb driver should prevent.
But if it somehow happens, then we will anyway run into problems calling `enif_release_resource(future)` twice.

File: c_src/main.c [hunk 2]
```diff
@@ -86,23 +86,16 @@ erlfdb_future_cb(FDBFuture* fdb_future, void* data)
     cancelled = future->cancelled;
     enif_mutex_unlock(future->cancel_lock);
```
Comment (qzhuyan) [inline c_src/main.c:87]: Stupid question, do we released the lock too early? 
image two processes are cancelling the future if it can happen?

Comment (savonarola) [reply c_src/main.c:87]: fdb allows double cancelling; it only forbids double destroy

Comment (qzhuyan) [reply c_src/main.c:87]: If FDB ensures the callback is executed exact once as the cancel could be triggered by FDB or by NIF.
and the deref of future NIF resource is only done (for NIF part) in the callback  https://github.com/emqx/couchdb-erlfdb/pull/22/files#diff-ceadfa165520612e5b6b1a255f7680bcaa9e2a67bf47677f6884fd1db3cb03f5L110 .
Then I think the future->cancel_lock is not needed at all?

**Review by qzhuyan — APPROVED** [review]

File: c_src/main.c (hunk 2 above)
Comment (qzhuyan) [reply c_src/main.c:87]: it is unrelated to this PR though.

Comment (savonarola) [reply c_src/main.c:87]: `future->cancelled` is independently set from two different nifs;  so, formally, we need synchronization

**Pull request merged**